src/profile-logic/process-profile.js

```diff
@@ -801,6 +803,56 @@ function _processCounters(
   );
 }
 
+/**
+ * Converts the Gecko profiler overhead into the processed format.
+ */
+function _processProfilerOverhead(
+  geckoProfile: GeckoProfile | GeckoSubprocessProfile,
+  // The overhead data is listed independently from the threads, so we need an index that
+  // references back into a stable list of threads. The threads list in the processing
+  // step is built dynamically, so the "stableThreadList" variable is a hint that this
+  // should be a stable and sorted list of threads.
+  stableThreadList: Thread[],
+  // The timing across processes must be normalized, this is the timing delta between
+  // various processes.
+  delta: Milliseconds
+): ProfilerOverhead | null {
+  const geckoProfilerOverhead: ?GeckoProfilerOverhead =
+    geckoProfile.profilerOverhead_UNSTABLE;
+  const mainThread = geckoProfile.threads.find(
+    thread => thread.name === 'GeckoMain'
+  );
+
+  if (!mainThread || !geckoProfilerOverhead) {
+    // Profiler overhad or a main thread weren't found, bail out, and return an empty array.
```

Comment:
```suggestion
    // Profiler overhead or a main thread weren't found, bail out, and return an empty array.
```

Reply:
Does the overhead data only exist on the GeckoMain? What happens when we don't record GeckoMain?

Reply:
Actually we are attaching this to `GeckoMain` because I followed the same implementation of `counters`. I was thinking that counters were doing the correct thing. We are displaying both counters and profilerOverhead under GeckoMain thread and if there is no GeckoMain thread, we don't display them. What do you think? If we want to change this, I think we should change counters as well.